### ISSUE TITLE
feature-request: release all functions

### ISSUE THREAD

**Issue by guybrush** [c9/smith#2]
what i want to do:

``` javascript
var rap = require('remoteagent-protocol')
var net = require('net')
var l = console.log
var port = ~~(Math.random()*50000)+10000
var server = net.createServer(function(s){
  l('S> C connected')
  var r = new rap.Remote(s,s,0)
  s.on('end',function(){
    l('S> C disconnected')
    // here is where we need to release all functions
  })
  r.emitRemote('init',{name:function(cb){cb('server')}})
  r.once('init',function(fns){
    setInterval(function(){
      // this will Error once the client has disconnected 
      // (ie. we cant write to the socket anymore)
      fns.name(function(n){l('S> remote-name: '+n)})
    },1000)
  })
})
server.listen(port,function(){
  l('S> listening')
  var client = net.connect(port,function(){
    var r = new rap.Remote(client,client,1)
    r.emitRemote('init',{name:function(cb){cb('client')}})
    r.once('init',function(fns){
      fns.name(function(n){
        l('C> remote-name: '+n)
        setTimeout(function(){client.end()},1000)
      })
    })
  })
})
```

**Comment by creationix** [comment]
Sorry for being so slow to get back to you.  The purpose of this project has shifted slightly. (It's more usable now!)

I changed the semantics of the function serializing too.  It now has zero function reference leaks by default.  There is no reason to release function handles automatically.  The new constraint is that anonymous functions can only be called once.  In order for repeating events to work, I've added named functions that can be called as many times as wanted.
